ci/check-documentation.py

```diff
@@ -29,22 +30,41 @@
     'tt', 'u', 'ul', 'var',
 ])
 
-def find_md_files():
-    thirdparty_dir = OMZ_ROOT / 'demos' / 'thirdparty'
+thirdparty_dir = OMZ_ROOT / 'demos' / 'thirdparty'
+
 
+def find_md_files():
     for path in OMZ_ROOT.glob('**/*.md'):
         if thirdparty_dir in path.parents: continue
         yield path
 
 def main():
     all_passed = True
+    
+    md_check_cases = (f'{OMZ_ROOT}{os.sep}models{os.sep}intel{os.sep}index.md', 
+                     f'{OMZ_ROOT}{os.sep}models{os.sep}public{os.sep}index.md', 
+                     f'{OMZ_ROOT}{os.sep}demos{os.sep}README.md')
 
     def complain(message):
         nonlocal all_passed
         all_passed = False
         print(message, file=sys.stderr)
 
     for md_path in sorted(find_md_files()):
+        check_md_links = False
+        readme_files = list()
+
+        if os.path.normpath(md_path) in md_check_cases:
+            check_md_links = True
+            path_folder = os.path.dirname(md_path)
+
+            md_paths = tuple(Path(path_folder).glob('*/*/README.md')) + \
+                       tuple(Path(path_folder).glob('*/README.md'))
```

Comment:
1. I don't think you actually need to glob here; we already do the globbing in `find_md_files`, so you can just save the result of that into a variable and then filter that variable. (Incidentally, this will also remove the need to filter out the files in `thirdparty`, because `find_md_files` already does that.)

2. Finding README files at only these specific depth levels may work now, but it might silently miss some files if we alter the layout of the demo/model directories. I would suggest to instead change the logic so that it finds all `README.md` files in the subtree whose parent directories don't contain any other `README.md` files (e.g. `demos/speech_recognition_demo/python/ctcdecode-numpy/README.md` should not be included, because `demos/speech_recognition_demo/python/README.md` exists).